lib/i18n_yaml_editor/store.rb

```diff
@@ -76,14 +76,10 @@ def create_missing_keys
         missing_locales.each {|locale|
           translation = key.translations.first
 
-          # this just replaces the locale part of the file name. should
-          # be possible to do in a simpler way. gsub, baby.
-          path = Pathname.new(translation.file)
-          dirs, file = path.split
-          file = file.to_s.split(".")
-          file[-2] = locale
-          file = file.join(".")
-          path = dirs.join(file).to_s
+          # this just replaces the locale part of the file name
+          path = translation.file
+              .sub(/(\/|\.)#{translation.locale}\.yml$/, "\\1#{locale}.yml")
+              .sub(/\/#{translation.locale}([^\/]+)\.yml$/, "/#{locale}\\1.yml")
```

Comment:
Find these regexes really tough to read, when what we're trying to do is remove the `-2` constant. How about:

```ruby
parts = File.basename(translation.file).split('.')
parts[parts.index(translation.locale)] = locale

path = "#{File.dirname(translation.file)/#{parts.join('.')}"
```

:grin: